### PR TITLE
Fix Greek Mythology references in compiler package

### PR DESCRIPTION
**Pull request by janekdb**
Improve documentation in some packages and correct some typos in these directories,

 - compiler
 - library

### CODE REVIEW

**Comment by SethTisue** [comment]
(CI failure is real)

**Comment by janekdb** [comment]
Fix this
```
[error] /home/jenkins/workspace/scala-2.12.x-validate-publish-core/src/compiler/scala/tools/nsc/typechecker/Implicits.scala:363: overriding value runDefinitions in class Typer of type Implicits.this.global.definitions.RunDefinitions;
[error]  value runDefinitions has weaker access privileges; it should not be private
[error]     private val runDefinitions = currentRun.runDefinitions
[error]                 ^
[error] one error found
```

**Comment by janekdb** [comment]
Compile error fixed. Should be plain sailing here on in.

image

File: src/compiler/scala/tools/nsc/typechecker/Implicits.scala [hunk 1]
```diff
@@ -360,8 +360,8 @@ trait Implicits {
     val undetParams = if (isView) Nil else context.outer.undetparams
     val wildPt = approximate(pt)
 
-    private val runDefintions = currentRun.runDefinitions
-    import runDefintions._
+    private val currentRunRunDefinitions = currentRun.runDefinitions
+    import currentRunRunDefinitions._
```
Comment (Ichoran) [inline src/compiler/scala/tools/nsc/typechecker/Implicits.scala]: Ow.  That is not a very friendly name.  If you're going to use so many letters, maybe at least explain why this whole thing is necessary `val stableRunDefsForImport = currentRun.runDefinitions`

File: src/compiler/scala/tools/nsc/typechecker/Namers.scala [hunk 2]
```diff
@@ -1348,7 +1348,7 @@ trait Namers extends MethodSynthesis {
 
       // Add a () parameter section if this overrides some method with () parameters
       val vparamSymssOrEmptyParamsFromOverride =
-        if (overridden != NoSymbol && vparamSymss.isEmpty && overridden.alternatives.exists(_.info.isInstanceOf[MethodType])) ListOfNil // NOTEL must check `.info.isInstanceOf[MethodType]`, not `.isMethod`!
+        if (overridden != NoSymbol && vparamSymss.isEmpty && overridden.alternatives.exists(_.info.isInstanceOf[MethodType])) ListOfNil // NOTE: Must check `.info.isInstanceOf[MethodType]`, not `.isMethod`!
```
Comment (Ichoran) [inline src/compiler/scala/tools/nsc/typechecker/Namers.scala]: `Must` should be `must` as it is a sentence fragment.  (If we're going to be really pedantic.)

File: src/compiler/scala/tools/nsc/typechecker/Implicits.scala (hunk 1 above)
Comment (SethTisue) [reply src/compiler/scala/tools/nsc/typechecker/Implicits.scala]: daDoRunRunDefinitions http://1.bp.blogspot.com/-xLVS4NExYCU/U9xUfmsNVcI/AAAAAAAABlQ/u6zx6uPJLis/s1600/shaun+4.jpg

Comment (dwijnand) [reply src/compiler/scala/tools/nsc/typechecker/Implicits.scala]: Ya running and ya running. But ya can't run away from yourself.

File: src/compiler/scala/tools/nsc/typechecker/Namers.scala (hunk 2 above)
Comment (janekdb) [reply src/compiler/scala/tools/nsc/typechecker/Namers.scala]: Now lowercase matching the majority case,

```
$ grep -rh -oE '(NOTE:.*)' src/|cut -c1-7|sort|uniq -c
      1 NOTE: `
      1 NOTE: *
      2 NOTE: a
      1 NOTE: A
      1 NOTE: b
     11 NOTE: c
      2 NOTE: C
      1 NOTE: d
      2 NOTE: e
      1 NOTE: E
      3 NOTE: f
      2 NOTE: g
      1 NOTE: h
      1 NOTE: i
      2 NOTE: I
      2 NOTE: m
      2 NOTE: o
      1 NOTE: q
      2 NOTE: r
      1 NOTE: s
      1 NOTE: S
      6 NOTE: t
      1 NOTE: T
      3 NOTE: U
      1 NOTE: V
      4 NOTE: w
      1 NOTE: W
```

File: src/compiler/scala/tools/nsc/typechecker/Implicits.scala (hunk 1 above)
Comment (janekdb) [reply src/compiler/scala/tools/nsc/typechecker/Implicits.scala]: @Ichoran - thanks for thinking about this. Improvement applied!

**Review by SethTisue — APPROVED** [review]
on behalf of the Nicolaas Govert de Bruijn estate, I thank you.

File: src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala [hunk 3]
```diff
@@ -117,13 +117,13 @@ trait TypeDiagnostics {
    */
   final def exampleTuplePattern(names: List[Name]): String = {
     val arity = names.length
-    val varPatterNames: Option[List[String]] = sequence(names map {
+    val varPatternNames: Option[List[String]] = sequence(names map {
```
Comment (l0rinc) [inline src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala:120]: minor: maybe remove `var` prefix if it's a `val` :)
```scala
val patternNames
```

**Review by l0rinc — APPROVED** [review]

File: src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala (hunk 3 above)
Comment (janekdb) [reply src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala:120]: In context I think the `val` is about variables so it's okay to leave as-is,

```
    val varPatterNames: Option[List[String]] = sequence(names map {
      case name if nme.isVariableName(name) => Some(name.decode)
      case _                                => None
    })
```
WDYT?

**Review by l0rinc — APPROVED** [review]

**Comment by janekdb** [comment]
Fixed conflicts.

**Review by l0rinc — APPROVED** [review]

**Comment by SethTisue** [comment]
thanks! looking forward to the PR that straightens out the scalac situation w/r/t qabbalism, the medieval era, etc.

**Pull request merged**